### PR TITLE
creation / management / burn ...

### PR DESCRIPTION
**Pull request by Menwitz**

### CODE REVIEW

File: Modules/core/BRPeer.c [hunk 1]
```diff
@@ -275,6 +280,95 @@ static int _PeerAcceptVerackMessage(BRPeer *peer, const uint8_t *msg, size_t msg
...
+    if (size == 0) {
+        peer_log(peer, "malformed assets message");
+        r = 0;
+    } else {
+        //TODO : assign name and len to new asset
+
+        nameLen = UInt64GetLE(&msg[off]);
+        off += sizeof(uint64_t);
+        
+        char name[nameLen];
```
Comment (TronBlack) [inline Modules/core/BRPeer.c:363]: Should this be nameLen +1

File: Modules/core/BRPeerManager.c [hunk 2]
```diff
@@ -2054,6 +2058,23 @@ const ChainParams *BRPeerManagerChainParams(BRPeerManager *manager) {
...
+void PeerManagerGetAssetData(BRPeerManager *manager, void *infoManager, char *assetName, size_t nameLen,
+                             void (*receivedAssetData)(void *info, BRAsset *asset)) {
+    
+    for (size_t i = array_count(manager->connectedPeers); i > 0; i--) {
+        BRPeer *peer = manager->connectedPeers[i - 1];
+        
+        if (BRPeerConnectStatus(peer) != BRPeerStatusConnected) continue;
+        
+        peer->assetCallbackInfo = infoManager;
+
```
Comment (TronBlack) [inline Modules/core/BRPeerManager.c]: Can we put this constraint back in?

```suggestion
if (BRPeerVersion(peer) < 70018) continue;
```

File: Modules/core/BRTransaction.h [hunk 3]
```diff
@@ -109,9 +109,9 @@ typedef struct {
 //    char name[31]; // MAX 31 Bytes
     size_t nameLen;
     uint64_t amount;     // 8 Bytes
-    int64_t unit;        // 1 Byte
-    int64_t reissuable;  // 1 Byte
-    int64_t hasIPFS;     // 1 Byte
+    int unit;        // 1 Byte
```
Comment (TronBlack) [inline Modules/core/BRTransaction.h]:
```suggestion
    int8_t unit;        // 1 Byte
```

File: Modules/core/BRTransaction.h [hunk 4]
```diff
@@ -109,9 +109,9 @@ typedef struct {
...
+    int reissuable;  // 1 Byte
```
Comment (TronBlack) [inline Modules/core/BRTransaction.h]:
```suggestion
    int8_t reissuable;  // 1 Byte
```

File: Modules/core/BRTransaction.h [hunk 5]
```diff
@@ -109,9 +109,9 @@ typedef struct {
...
+    int hasIPFS;     // 1 Byte
```
Comment (TronBlack) [inline Modules/core/BRTransaction.h]:
```suggestion
    int8_t hasIPFS;     // 1 Byte
```

Comment (TronBlack) [reply Modules/core/BRTransaction.h]: Reason: int may be different sizes depending on the bitness of the machine being used to compile.  int8_t should be a single 8-bit byte regardless of compiler or bitness of the machine.

File: Modules/core/BRWallet.c [hunk 6]
```diff
@@ -620,6 +709,21 @@ BRTransaction *BRWalletCreateTxForOutputs(BRWallet *wallet, const BRTxOutput *ou
         transaction->asset->nameLen = asset->nameLen;
         transaction->asset->amount = asset->amount;
         transaction->asset->type = asset->type;
+        transaction->asset->reissuable = asset->reissuable;
+        transaction->asset->unit = asset->unit;
+        transaction->asset->hasIPFS = asset->hasIPFS;
+        if(transaction->asset->hasIPFS == 1)
```
Comment (TronBlack) [inline Modules/core/BRWallet.c]: This seems backwards.
```suggestion
        if(asset->hasIPFS == 1)
```

File: Modules/core/BRWallet.c [hunk 7]
```diff
@@ -620,6 +709,21 @@ BRTransaction *BRWalletCreateTxForOutputs(BRWallet *wallet, const BRTxOutput *ou
...
+            strcpy(asset->IPFSHash, transaction->asset->IPFSHash);
```
Comment (TronBlack) [inline Modules/core/BRWallet.c]: This seems backward
```suggestion
            strcpy(transaction->asset->IPFSHash, asset->IPFSHash);
```

File: Modules/core/ChainParams.h [hunk 8]
```diff
@@ -29,13 +29,13 @@
 #include "BRSet.h"
 #include <assert.h>
 
-static const int64_t COIN = 100000000; // todo look into the amount! 100M not 1M corbies
+static const int64_t COIN = 100000000; // BMEX TODO: look into the amount! 100M not 1M corbies
```
Comment (TronBlack) [inline Modules/core/ChainParams.h]: Corbies are 1/100,00,000 of a RVN  10^-8

File: Modules/core/BRPeer.c (hunk 1 above)
Comment (Menwitz) [reply Modules/core/BRPeer.c:363]: I will have to remove it anyway, on iOS I play on asset!=nil

File: Modules/core/BRPeerManager.c (hunk 2 above)
Comment (Menwitz) [reply Modules/core/BRPeerManager.c]: can we use this for debug
        if (BRPeerVersion(peer) >= 70018) {
            BRPeerSendGetAsset(peer, assetName, nameLen, receivedAssetData);
        } else
            peer_log(peer, "node doesn't support Assets MSG Protocol");

File: Modules/core/BRTransaction.h (hunk 3 above)
Comment (Menwitz) [reply Modules/core/BRTransaction.h]: this is an old commit, I changed those to int:
    int unit;      
    int reissuable;
    int hasIPFS;     
for swift to C interfacing but I still manage to add them as varints

File: Modules/core/BRPeerManager.c (hunk 2 above)
Comment (TronBlack) [reply Modules/core/BRPeerManager.c]: Sure

File: Modules/core/BRTransaction.h (hunk 3 above)
Comment (TronBlack) [reply Modules/core/BRTransaction.h]: They shouldn't be int either.  They should be specified by number of bits.

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Review by cryptelius — APPROVED** [review]

**Pull request merged**